### DISCUSSION TITLE
Remap keys for telescope.actions

### DISCUSSION THREAD

**Discussion by oleynikandrey** [LazyVim/LazyVim#4781] [Q&A] [answered]
Hello, I'm trying to remap keys for scrolling in Telescope.

As Telescope's maintainer said [here](https://github.com/nvim-telescope/telescope.nvim/issues/2634#issuecomment-1666070443), this could be done like this.
```lua
require("telescope").setup({
  defaults = {
    -- .. other settings
    mappings = {
      i = {
        ["<C-a>"] = actions.results_scrolling_left,
        ["<C-e>"] = actions.results_scrolling_right,
      }
    }
  }
})
```

Where should I put the setup invocation?

I tried putting the following config in `plugins/telescope.lua`, but without success.
```lua
return {
  "nvim-telescope/telescope.nvim",
  config = function()
    local actions = require("telescope.actions")
    require("telescope").setup({
      defaults = {
        mappings = {
          i = {
            ["<D-C-h>"] = actions.results_scrolling_left,
            ["<D-C-l>"] = actions.results_scrolling_right,
          },
        },
      },
    })
  end,
}
```

**Comment by dpetka2001** [comment]
```lua
return {
  {
    "nvim-telescope/telescope.nvim",
    opts = function(_, opts)
      local actions = require("telescope.actions")

      opts.defaults = vim.tbl_deep_extend("force", opts.defaults, {
        mappings = {
          i = {
            ["<C-a>"] = actions.results_scrolling_left,
            ["<C-e>"] = actions.results_scrolling_right,
          },
        },
      })
    end,
  },
}
```

**Reply by oleynikandrey** [reply]
Unfortunately, it does not work. I receive the following error:

```
Failed to run `config` for telescope.nvim

vim/shared.lua:0: after the second argument: expected table, got nil

# stacktrace:
  - vim/shared.lua:0 _in_ **validate**
  - vim/shared.lua:0 _in_ **tbl_deep_extend**
  - ~/.config/nvim/lua/plugins/telescope.lua:5 _in_ **values**
  - ~/.config/nvim/lua/config/lazy.lua:17
  - ~/.config/nvim/init.lua:2
```

`opts` is empty table.

**Reply by dpetka2001** [reply]
You didn't copy it well it seems. I tested and it's working.

**Reply by oleynikandrey** [reply]
I have copied it as is and saw the same error.

**Reply by dpetka2001** [reply]
Pls post a link on github with your dotfiles. The error is about `tbl_deep_extend` after second argument not being a table. But in my example it is a table.

**Reply by oleynikandrey** [reply]
Sorry, I don't have my dot files on GitHub. As I said before, the `opts` variable I get is `{}`, so it has no `defaults` key.

**Reply by folke** [reply]
You're fully overwriting the telescope somewhere or you're not using LazyVim?

**Reply by oleynikandrey** [reply]
I have uploaded all my configs [here](https://github.com/oleynikandrey/lazyvim/tree/main/nvim).

**Reply by dpetka2001** [reply]
I don't see any configuration about Telescope. This doesn't help at all if i can't see what code you have for Telescope to see where the error is. Where's the `telescope.lua` file that was in the error you showed?

**Reply by oleynikandrey** [reply]
File the same as you provided. Pushed it to the repo. Please try again.

**Reply by folke** [reply, marked as answer]
You're using `lazyvim.plugins.extras.editor.fzf"`, so yes in that case telescope won't be configured....

You are not using telescope

**Reply by oleynikandrey** [reply]
Oh, I thought that fzf is a terminal fuzzy finder and Telescope just uses it.
Thank you for help.
